### PR TITLE
add sonar-scanner formula, replacement for sonar-runner

### PR DESCRIPTION
**Pull request by sv99**
sonar-runner deprecated

WARN: sonar-runner script is deprecated. Please use sonar-scanner instead.

### CODE REVIEW

File: Formula/sonar-scanner.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class SonarScanner < Formula
+  desc "Launcher to analyze a project with SonarQube"
+  homepage "http://docs.sonarqube.org/display/SONAR/Analyzing+with+SonarQube+Scanner"
```
Comment (DomT4) [inline Formula/sonar-scanner.rb]: Looks like this redirects to `http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner` so we may as well use that.

File: Formula/sonar-scanner.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def install
+    # Remove windows files
+    rm_rf Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+    bin.write_exec_script libexec/"bin/sonar-scanner"
+  end
+
+  def caveats; <<-EOS.undent
+      If this is your first install, you should adjust the default configuration:
```
Comment (DomT4) [inline Formula/sonar-scanner.rb]: Use a two space indent rather than four, thanks!

File: Formula/sonar-scanner.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        #{libexec}/conf/sonar-scanner.properties
```
Comment (DomT4) [inline Formula/sonar-scanner.rb]: `opt_libexec` produces a slightly nicer output.

File: Formula/sonar-scanner.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+      after that you should also add a new enviroment variable:
+        SONAR_SCANNER_HOME=#{libexec}
```
Comment (DomT4) [inline Formula/sonar-scanner.rb]: We should env script this ideally. Something like [this](https://github.com/Homebrew/homebrew-core/blob/4fdd781d53d3fccb35a3c2be286b9849ff807adc/Formula/chapel.rb#L31).

File: Formula/sonar-scanner.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      EOS
+  end
+
+  test do
+    system "#{bin}/sonar-scanner", "-h"
```
Comment (DomT4) [inline Formula/sonar-scanner.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by DomT4 — CHANGES_REQUESTED** [review]

File: Formula/sonar-scanner.rb (hunk 5 above)
Comment (sv99) [inline Formula/sonar-scanner.rb]: I simply rewrite existing formula. sonar-runner - deprecated by sonarcube.

**Comment by sv99** [comment]
updated

File: Formula/sonar-scanner.rb (hunk 3 above)
Comment (sv99) [inline Formula/sonar-scanner.rb]: --version show full path to the configuration file:
```
sonar-scanner --version
...
INFO: Scanner configuration file: /usr/local/Cellar/sonar-scanner/2.8/libexec/conf/sonar-scanner.properties
...
```

File: Formula/sonar-scanner.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    If this is your first install, you should adjust the default configuration:
+      #{libexec}/conf/sonar-scanner.properties
+    EOS
+  end
+
+  test do
+    ENV["SONAR_SCANNER_HOME"] = libexec
```
Comment (dunn) [inline Formula/sonar-scanner.rb]: This shouldn't be necessary, since the script in `bin` is an env wrapper.

File: Formula/sonar-scanner.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  bottle :unneeded
+
+  def install
+    # Remove windows files
+    rm_rf Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+    bin.install libexec/"bin/sonar-scanner"
+
+    # sonar-sanner uses this ENV to work out where to install.
+    ENV["SONAR_SCANNER_HOME"] = libexec
```
Comment (dunn) [inline Formula/sonar-scanner.rb]: You've already installed, so is this necessary?

File: Formula/sonar-scanner.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    bin.env_script_all_files libexec/"bin/", :SONAR_SCANNER_HOME => libexec
+  end
+
+  def caveats; <<-EOS.undent
+    If this is your first install, you should adjust the default configuration:
+      #{libexec}/conf/sonar-scanner.properties
```
Comment (dunn) [inline Formula/sonar-scanner.rb]: Is this Homebrew-specific information? If not, please remove it!

File: Formula/sonar-scanner.rb (hunk 7 above)
Comment (sv99) [inline Formula/sonar-scanner.rb]: SONAR_SCANNER_HOME - without this variable - runtime error

File: Formula/sonar-scanner.rb (hunk 8 above)
Comment (sv99) [inline Formula/sonar-scanner.rb]: This is homebrew specific path of the global configuration file sonar-scanner.properties

File: Formula/sonar-scanner.rb (hunk 7 above)
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: If it's a runtime error then this won't be set at runtime.

File: Formula/sonar-scanner.rb (hunk 8 above)
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: Just remove the caveats then and install them into `etc.install` like other configuration files.

Comment (sv99) [inline Formula/sonar-scanner.rb]: sonar-scanner --version show: 
INFO: Scanner configuration file: /usr/local/Cellar/sonar-scanner/2.8/libexec/conf/sonar-scanner.properties

?? not to the etc.install ??

**Comment by sv99** [comment]
updated

File: Formula/sonar-scanner.rb (hunk 8 above)
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: Ideally we'd convince it to put the file elsewhere. If not we can `etc.install` and use a symlink so both locations still work.

Comment (sv99) [inline Formula/sonar-scanner.rb]: this path used in the script from SONAR_SCANNER_HOME. How can i add link -  etc.install copy script to the /usr/local/etc?

File: Formula/sonar-scanner.rb (hunk 7 above)
Comment (dunn) [inline Formula/sonar-scanner.rb]: The env wrapper script should take care of that, but it's not being run in the `def install` block, so it shouldn't be necessary there.

File: Formula/sonar-scanner.rb (hunk 8 above)
Comment (dunn) [inline Formula/sonar-scanner.rb]:
```ruby
etc.install "conf/sonar-scanner.properties"
ln_s etc/"conf/sonar-scanner.properties", libexec/"conf/sonar-scanner.properties"

File: Formula/sonar-scanner.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+  def install
+    # Remove windows files
+    rm_rf Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+    bin.install libexec/"bin/sonar-scanner"
+    etc.install libexec/"conf/sonar-scanner.properties"
+    ln_s etc/"sonar-scanner.properties", libexec/"conf/sonar-scanner.properties"
+
+    # sonar-sanner uses this ENV to work out where to install.
```
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: Can remove this variable.

File: Formula/sonar-scanner.rb [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  desc "Launcher to analyze a project with SonarQube"
+  homepage "http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner"
+  url "https://sonarsource.bintray.com/Distribution/sonar-scanner-cli/sonar-scanner-2.8.zip"
+  sha256 "0295365a7e5d4499ec6b46cb6c70f3fa127159b58b73930f675acd0897a6b350"
+  head "https://github.com/SonarSource/sonar-scanner-cli.git"
+
+  bottle :unneeded
+
+  def install
+    # Remove windows files
```
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: Can remove this comment.

File: Formula/sonar-scanner.rb (hunk 9 above)
Comment (MikeMcQuaid) [inline Formula/sonar-scanner.rb]: Comment, sorry, not variable.

File: Formula/sonar-scanner.rb (hunk 10 above)
Comment (sv99) [inline Formula/sonar-scanner.rb]: updated

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `sonar-scanner 2.8 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request closed without merging**